src/main/java/io/confluent/kafka/connect/datagen/DatagenConnector.java

```diff
@@ -37,11 +43,33 @@
   private static Logger log = LoggerFactory.getLogger(DatagenConnector.class);
   private DatagenConnectorConfig config;
   private Map<String, String> props;
+  private Generator generator;
 
   @VisibleForTesting
   static final String SCHEMA_SOURCE_ERR =
       "Must set exactly one of " + String.join(", ", DatagenConnectorConfig.schemaSourceKeys());
 
+  static final String QUICKSTART_ERR =
+           DatagenConnectorConfig.QUICKSTART_CONF + " must be one out of "
+                   + String.join(",", DatagenTask.Quickstart.configValues);
+
+  static final String QUICKSTART_SCHEMA_ERR =
+          "The provided schema in '" + DatagenConnectorConfig.QUICKSTART_CONF + "' is invalid";
+
+  static final String SCHEMA_STRING_ERR =
+          "The provided schema in '" + DatagenConnectorConfig.SCHEMA_STRING_CONF + "' is invalid";
+
+  static final String SCHEMA_FILE_ERR =
+          "The provided schema in '" + DatagenConnectorConfig.SCHEMA_FILENAME_CONF + "' is invalid";
+
+  static final String SCHEMA_FILE_NOT_FOUND =
+          "Unable to read the schema file provided in '"
+                  + DatagenConnectorConfig.SCHEMA_FILENAME_CONF + "'";
+
+  static final String SCHEMA_KEY_FIELD_NOT_FOUND =
+          "The schema does not contain the field provided in '"
+                  + DatagenConnectorConfig.SCHEMA_KEYFIELD_CONF + "'";
```

Comment:
I personally find that this style of relegating error messages to constants at the top of a file does more harm than good. It's easier to read when the error strings are defined closer to where they're actually used, and although it does make it easier to unit test the content of error messages, in practice, that doesn't really buy that much (since that level of granularity is rarely necessary anyways) and isn't worth the drop in readability IMO.

Reply:
Hm not sure how much that buys us in terms of readability considering you'd still have to navigate to the declaration or the top of the method (which is fairly large here). 

> since that level of granularity is rarely necessary anyways
An example where this is useful - [here](https://github.com/confluentinc/kafka-connect-datagen/pull/84/files#diff-138280df551c89fab27c3b0bf6607109feb0b2710594d51bf4ca19bc00ca4061R177). We can make sure that the validation error returned is that the schema provided in the file is invalid and NOT that the schema file couldn't be read at all

Reply:
Why does the declaration need to be at the top of the method? And why are these assertions on error messages necessary? I don't know of any production code bases where this type of effort is expended, and considering this connector is a demo tool, I just don't see the point in expending more energy and making development slower and harder on this project for very little benefit.

Reply:
> And why are these assertions on error messages necessary

They aren't necessary per se, but it's nice to be able to test that the correct validation error is being returned in each scenario. I'm okay with both ways, so I'll go ahead and change the tests to just verify the number of error messages instead of matching the exact error string

Reply:
I've gone ahead and removed these static declarations in favor of using them directly in the validator classes